### ISSUE TITLE
Thoughts on trades-based operation

### ISSUE THREAD

**Issue by jorisw** [magic8bot/magic8bot#262]
Any trading bot such as magic8bot heavily relies on backfills, i.e. retrieving trade history from exchanges, to warm up indicators with. magic8bot also heavily relies on ccxt, the library that unifies various crypto exchanges into a single unified API.

The manual of ccxt [stresses](https://ccxt.readthedocs.io/en/latest/manual.html#public-trades) that in general you should not rely on exchanges being able or willing to supply trade history far into the past. Exchanges that don't offer this, generally do offer long candlestick (OHLCV) histories, that basically aggregate the same data into 1m to 1M periods.

Long story short, the reliance of magic8bot on trades, rather than candles for historical data, seems like quite an anti pattern.

I'm curious to know if current maintainers (if any) of magic8bot concur.

**Comment by notVitaliy** [comment]
The reason for aggregating trade data instead of candle data is pretty simple. Trades can be grouped into time-buckets which can be converted into any period of candles on the fly. The same thing can probably be done with 1m candles as well.

This was a design decision that was made very early into the project and was really only tested with coinbase pro and binance.

I don't remember exactly why, as it was a long time ago, but I believe that either one of those (or both) did not offer candle data at that time.

In theory, the time-bucket function can be changed so that it converts from 1m candles to the desired period. The backfill can be modified to import either 1m candles or trades, then convert them to 1m candles before storing in mongo, depending on which data the exchange provides.

**Comment by jorisw** [comment]
> the time-bucket function can be changed so that it converts from 1m candles to the desired period

Indeed that is what I'm doing with my project, except it (unfortunately) is not based on magic8bot.
